### PR TITLE
Replace askama with rinja

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Another askama maintainer and myself forked it and named the fork rinja. The whole difference is explained in this [blog post](https://blog.guillaume-gomez.fr/articles/2024-07-16+docs.rs+switching+jinja+template+framework+from+tera+to+rinja) (not publicly released yet, waiting for docs.rs to deploy the new version using it before). But in short, rinja got a lot of improvements and compiles faster, so I think it's definitely worth it to use it in rustdoc as well.

r? @notriddle

### CODE REVIEW

File: src/librustdoc/html/templates/STYLE.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 # Style for Templates
 
 This directory has templates in the [Tera templating language][teradoc], which is very
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/templates/STYLE.md:3]: They're supposed (tera, askama and rinja) to be compatible with `jinja`. Askama and rinja share the same language too. I'll reword this though. Sounds weird.

**Comment by GuillaumeGomez** [comment]
Reworded the document. Lot of strange things in there.

**Comment by GuillaumeGomez** [comment]
@Kobzol Considering it should only impact compilation of rustdoc itself, not sure it's visible in the benches. Oh well, at least it confirmed that it didn't impact things negatively. :)

**Comment by Kobzol** [comment]
Oh, I thought that this is also affecting the performance of `cargo doc`/`rustdoc`. In any case, at least we know that perf is neutral, as you said.

**Comment by GuillaumeGomez** [comment]
Well, let's wait for @notriddle to give a look then. ;)

**Comment by GuillaumeGomez** [comment]
I'll update the priority so I don't have to rebase too often...

@bors r+ p=1

**Pull request merged**